### PR TITLE
Upgrade eigen to 3.4.0

### PR DESCRIPTION
**Pull request by vaithak**
After this, the issue #105 (replacing Eigen pointers with iterators) can be worked upon by anyone. IMO, it can be given a label of `hacktoberfest`.

### CODE REVIEW

File: test/volume_cb_zonotopes.cpp [hunk 1]
```diff
@@ -137,15 +138,15 @@ void call_test_uniform_generator(){
     exact_vol = exact_zonotope_vol<NT>(P);
     test_volume_hpoly(P,
                       0,
-                      5.7603e+20,
```
Comment (vissarion) [inline test/volume_cb_zonotopes.cpp:140]: any idea of what have changed in Eigen and there are those changes in test results ?

**Review by vissarion — APPROVED** [review]
Thanks!

File: test/volume_cb_zonotopes.cpp (hunk 1 above)
Comment (vaithak) [reply test/volume_cb_zonotopes.cpp:140]: not sure, my guess is that these are due to numerical instability for skinny objects.

Comment (vissarion) [reply test/volume_cb_zonotopes.cpp:140]: but the process is deterministic (in the tests), they probably changed some algorithms in eigen that volesti uses

**Pull request merged**